packages/core/src/components/essentials/threshold.ts

```diff
@@ -0,0 +1,166 @@
+
+		const scale = this.services.cartesianScales.getScaleByPosition(axisPosition);
+		const isVertical = [AxisPositions.LEFT, AxisPositions.RIGHT].includes(axisPosition);
+		const scaleType = this.services.cartesianScales.getScaleTypeByPosition(axisPosition);
+		const mainXScale = this.services.cartesianScales.getMainXScale();
+		const mainYScale = this.services.cartesianScales.getMainYScale();
+		const isScaleTypeLabels = scaleType === ScaleTypes.LABELS;
+		const [xScaleStart, xScaleEnd] = mainXScale.range();
+		const [yScaleEnd, yScaleStart] = mainYScale.range();
+
+		if (isVertical) {
+			const y = scale(value) + (isScaleTypeLabels ? (scale.step() / 2) : 0) - yScaleStart;
+			this.threshold.attr("transform", `translate(${xScaleStart}, ${y})`);
+			thresholdLine.attr("x2", xScaleEnd - xScaleStart);
+			thresholdRect.attr("width", xScaleEnd - xScaleStart);
+		} else {
+			const x = scale(value) + (isScaleTypeLabels ? (scale.step() / 2) : 0);
+			this.threshold.attr("transform", `translate(${x}, 0)`);
+			thresholdLine.attr("y2", yScaleEnd);
+			thresholdRect.attr("width", yScaleEnd - yScaleStart);
+			thresholdRect.classed("rotate", true);
+		}
+
+		const self = this;
+		this.services.events.addEventListener(Events.Threshold.SHOW, e => {
+			const hovered = e.detail.hoveredElement.node();
+			if (hovered === self.threshold) {
+				this.setThresholdLabelPosition();
+				this.label.classed("hidden", false);
+			}
+		});
+
+		this.services.events.addEventListener(Events.Threshold.HIDE, e => {
+			this.label.classed("hidden", true);
+		});
+
+		this.appendThresholdLabel();
+
+		this.addEventListeners();
+	}
+
+	getFormattedValue() {
+		const { value, axisPosition } = this.configs;
+		const options = Tools.getProperty(this.model.getOptions());
+		const scaleType = this.services.cartesianScales.getScaleTypeByPosition(axisPosition);
+
+		if (scaleType === ScaleTypes.TIME) {
+			const isVertical = [AxisPositions.LEFT, AxisPositions.RIGHT].includes(axisPosition);
+			const mainXScale = this.services.cartesianScales.getMainXScale();
+			const mainYScale = this.services.cartesianScales.getMainYScale();
+			const scale = isVertical ? mainYScale : mainXScale;
+
+			const timeScaleOptions = Tools.getProperty(options, "timeScale");
+			const timeInterval = computeTimeIntervalName(scale.ticks());
+			return formatTick(value, 0, timeInterval, timeScaleOptions);
+		}
+
+		return value.toLocaleString("en");
+	}
+
+	appendThresholdLabel() {
+		const { value, valueFormatter, fillColor, label = "Threshold" } = this.configs;
+		const holder = select(this.services.domUtils.getHolder());
+		const formattedValue = valueFormatter ? valueFormatter(value) : this.getFormattedValue();
+
+		this.label = DOMUtils.appendOrSelect(holder, `div.${this.thresholdClass}--label.${this.thresholdIdentifierClass}`);
+		this.label.html(`${label}: ${formattedValue}`);
+		this.label.classed("hidden", true);
+		this.label.style("background-color", fillColor);
```

Comment:
```suggestion
		this.label.html(`${label}: ${formattedValue}`)
				.classed("hidden", true)
				.style("background-color", fillColor);
```

just a suggestion, in most places we tend to use chain syntax where applicable